### PR TITLE
Kernel: Kernel_Drivers:Display: Device Tree Node: Added comments about necessity of HPD pin for J721S2

### PR DESCRIPTION
**Pull request by h-shenoy**
Add comments about necessity of HPD pin for J721S2.

### CODE REVIEW

**Comment by h-shenoy** [comment]
@uditkumarti @manorit2001

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst [hunk 1]
```diff
@@ -362,7 +362,12 @@ Device Tree Node
 Documentation for tidss device tree node and its properties can be found in linux kernel device tree bindings in below directory
 ``Documentation/devicetree/bindings/display/ti/``. Seperate binding files are present for different version of the DSS controller.
 
+.. ifconfig:: CONFIG_part_variant in ('J721S2')
 
+        - **MCASP1_ACLKX.DP0_HPD** is required for DP. Although it is not connected, it cannot be used as GPIO.
```
Comment (uditkumarti) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: MCASP1_ACLKX.DP0_HPD does not mention full scope of pin. 
You may use Ball number AA24..Something like 
Ball number AA24 is used for HPD in mux mode 3, irrespective of this is connected as HPD or not, this pin can not be used for other purpose, if DP0 is used.

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst [hunk 2]
```diff
@@ -362,7 +362,12 @@ Device Tree Node
...
+        - This pin is normally unused, but DPI to DP bridges from third parties might be referencing this pin regardless of whether it is connected in
+	  hardware or software. Setting this pin as GPIO may result in undefined behavior.
```
Comment (uditkumarti) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:369]: Do you mean DP0 is used with SOC bridge here , to generate DSI signals ?

**Review by uditkumarti — CHANGES_REQUESTED** [review]

**Comment by h-shenoy** [comment]
@StaticRocket

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst (hunk 1 above)
Comment (nmenon) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst]: be careful about Ball naming -> you have to add the package type with the ball name

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst [hunk 3]
```diff
@@ -362,7 +362,10 @@ Device Tree Node
 Documentation for tidss device tree node and its properties can be found in linux kernel device tree bindings in below directory
 ``Documentation/devicetree/bindings/display/ti/``. Seperate binding files are present for different version of the DSS controller.
 
+.. ifconfig:: CONFIG_part_variant in ('J721S2')
 
+        - Pin designated as HPD for DP should not be used for other purpose, irrespective of whether it is connected as HPD or not.
```
Comment (StaticRocket) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: > The pin designated as HPD for DP should not be used for other purpose, irrespective of whether it is connected to HPD.

Indentation here is also a bit odd. RST standard is 3 spaces. This file is currently using 4 spaces, but this line is using 8.

**Review by StaticRocket — CHANGES_REQUESTED** [review]

**Review by StaticRocket — CHANGES_REQUESTED** [review]
This new commit just adds actual whitespace errors. There needs the be a new line separating a directive from it's content and a newline to end a content block.

Having any whitespace on either of these lines confuses the parser as to what is content and what the end of the content block is.

**Review by StaticRocket — APPROVED** [review]

**Review by uditkumarti — APPROVED** [review]

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst [hunk 4]
```diff
@@ -362,6 +362,9 @@ Device Tree Node
 Documentation for tidss device tree node and its properties can be found in linux kernel device tree bindings in below directory
 ``Documentation/devicetree/bindings/display/ti/``. Seperate binding files are present for different version of the DSS controller.
 
+.. ifconfig:: CONFIG_part_variant in ('J721S2')
+
+   - Pin designated as HPD for DP should not be used for other purpose, irrespective of whether it is connected as HPD or not.
```
Comment (nmenon) [inline source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: is this covered in TRM? if so, point out chapter. DO NOT make s/w SDK documentation the goto place for careabouts of the SoC.

Comment (StaticRocket) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: @h-shenoy could you verify this and resolve this conversation if it is actually covered in the TRM?

Comment (h-shenoy) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: No, I did not find information regarding HPD pin connection status in TRM , let me know if information regarding the can be made part of SDK ?

Comment (nmenon) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: ideally, you wont document SoC features in S/w documentation - you should file a internal ticket asking it to be documented in TRM and discuss with applications team how to handle it best.

**Review by cshilwant — CHANGES_REQUESTED** [review]
Rebase your PR on master for checkers to start their job

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst (hunk 4 above)
Comment (h-shenoy) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: @uditkumarti , should this be documented in TRM and not in SDK docs as mentioned in ticket?

**Review by StaticRocket — CHANGES_REQUESTED** [review]
Please update in accordance with contribution guidelines

File: source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst (hunk 4 above)
Comment (StaticRocket) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: @uditkumarti

Comment (uditkumarti) [reply source/linux/Foundational_Components/Kernel/Kernel_Drivers/Display/DSS7.rst:367]: needs to be part of TRM . Please reject this PR

**Comment by cshilwant** [comment]
https://github.com/TexasInstruments/processor-sdk-doc/pull/2#discussion_r1867044628

**Pull request closed without merging**